server/src/test/java/org/apache/uniffle/server/ShuffleServerTest.java

```diff
@@ -79,4 +75,61 @@ public void startTest() {
     }
 
   }
+
+  public void decommissionTest(boolean shutdown) throws Exception {
+    ShuffleServerConf serverConf = createShuffleServerConf();
+    serverConf.set(SERVER_DECOMMISSION_CHECK_INTERVAL, 1000L);
+    serverConf.set(SERVER_DECOMMISSION_SHUTDOWN, shutdown);
+    ShuffleServer shuffleServer = new ShuffleServer(serverConf);
+    shuffleServer.start();
+    assertEquals(ServerStatus.NORMAL_STATUS, shuffleServer.getServerStatus());
+    // Shuffle server is not decommissioning, but we can also cancel it.
+    shuffleServer.cancelDecommission();
+    ShuffleTaskManager shuffleTaskManager = shuffleServer.getShuffleTaskManager();
+    String appId = "decommissionTest_appId_" + shutdown;
+    shuffleTaskManager.registerShuffle(appId, 0, Lists.newArrayList(), new RemoteStorageInfo("/tmp"), "");
+    shuffleServer.decommission();
+    assertEquals(ServerStatus.DECOMMISSIONING, shuffleServer.getServerStatus());
+    // Shuffle server is decommissioning, but we can also decommission it again.
+    shuffleServer.decommission();
+    shuffleServer.cancelDecommission();
+    shuffleTaskManager.removeResources(appId);
+    // Wait for 2 seconds, make sure cancel command is work.
+    Thread.sleep(2000);
+    assertEquals(ServerStatus.NORMAL_STATUS, shuffleServer.getServerStatus());
+    shuffleServer.decommission();
+    if (shutdown) {
+      Awaitility.await().timeout(10, TimeUnit.SECONDS).until(
+          () -> !shuffleServer.isRunning());
+    } else {
+      Awaitility.await().timeout(10, TimeUnit.SECONDS).until(
+          () -> ServerStatus.DECOMMISSIONED.equals(shuffleServer.getServerStatus()));
+      assertEquals(true, shuffleServer.isRunning());
+      shuffleServer.stopServer();
+    }
+  }
+
+  @Test
+  public void notShutDownAfterDecommissionTest() throws Exception {
+    decommissionTest(false);
+  }
+
+  @Test
+  public void shutDownAfterDecommissionTest() throws Exception {
+    decommissionTest(true);
+  }
+
+  private ShuffleServerConf createShuffleServerConf() throws Exception {
+    ShuffleServerConf serverConf = new ShuffleServerConf();
+    serverConf.setInteger(ShuffleServerConf.RPC_SERVER_PORT, 9527);
+    serverConf.setString(ShuffleServerConf.RSS_STORAGE_TYPE, StorageType.LOCALFILE.name());
+    serverConf.setBoolean(ShuffleServerConf.RSS_TEST_MODE_ENABLE, true);
+    serverConf.setInteger(ShuffleServerConf.JETTY_HTTP_PORT, 9528);
+    serverConf.setString(ShuffleServerConf.RSS_COORDINATOR_QUORUM, "localhost:0");
+    serverConf.set(ShuffleServerConf.RSS_STORAGE_BASE_PATH, Arrays.asList("/home/fengxj/tmp111"));
```

Comment:
Use `@TempDir` to create a temp dir for it.

```suggestion
    serverConf.set(ShuffleServerConf.RSS_STORAGE_BASE_PATH, Arrays.asList());
```

Reply:
Let's keep the original path.